core/src/main/java/org/apache/accumulo/core/logging/ScanUserDataLogger.java

```diff
@@ -0,0 +1,169 @@
+    if (classLogger != null) {
+      classLogger.debug(msg, objects);
+    }
+    if (!LOG.isDebugEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.debug(msg, objects);
+  }
+
+  public static void logInfo(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.info(msg, objects);
+    }
+    if (!LOG.isInfoEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.info(msg, objects);
+  }
+
+  public static void logWarn(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.warn(msg, objects);
+    }
+    if (!LOG.isWarnEnabled()) {
+      return;
+    }
+    USER_DATA.set(userData);
+    LOG.warn(msg, objects);
+  }
+
+  public static void logError(Logger classLogger, String userData, String msg, Object... objects) {
+    if (classLogger != null) {
+      classLogger.error(msg, objects);
+    }
+    USER_DATA.set(userData);
+    LOG.error(msg, objects);
+  }
+
+  @Override
+  public boolean isTraceEnabled() {
+    return LOG.isTraceEnabled();
+  }
+
+  @Override
+  public boolean isTraceEnabled(Marker marker) {
+    return LOG.isTraceEnabled(marker);
+  }
+
+  @Override
+  public boolean isDebugEnabled() {
+    return LOG.isDebugEnabled();
+  }
+
+  @Override
+  public boolean isDebugEnabled(Marker marker) {
+    return LOG.isDebugEnabled(marker);
+  }
+
+  @Override
+  public boolean isInfoEnabled() {
+    return LOG.isInfoEnabled();
+  }
+
+  @Override
+  public boolean isInfoEnabled(Marker marker) {
+    return LOG.isInfoEnabled(marker);
+  }
+
+  @Override
+  public boolean isWarnEnabled() {
+    return LOG.isWarnEnabled();
+  }
+
+  @Override
+  public boolean isWarnEnabled(Marker marker) {
+    return LOG.isWarnEnabled(marker);
+  }
+
+  @Override
+  public boolean isErrorEnabled() {
+    return LOG.isErrorEnabled();
+  }
+
+  @Override
+  public boolean isErrorEnabled(Marker marker) {
+    return LOG.isErrorEnabled(marker);
+  }
+
+  @Override
+  protected String getFullyQualifiedCallerName() {
+    return null;
+  }
+
+  @Override
+  @SuppressFBWarnings(value = "RV_RETURN_VALUE_IGNORED",
+      justification = "Return value is self, using fluent object in non-fluent manner")
+  protected void handleNormalizedLoggingCall(Level level, Marker marker, String messagePattern,
+      Object[] arguments, Throwable throwable) {
+
+    final String userData = USER_DATA.get();
+    USER_DATA.set(null);
+
+    LoggingEventBuilder builder = LOG.atLevel(level);
+    if (marker != null) {
+      builder.addMarker(marker);
+    }
+    if (throwable != null) {
+      builder.setCause(throwable);
+    }
+    builder.setMessage("({}) " + messagePattern);
+    builder.addArgument(userData);
+    if (arguments != null) {
+      for (Object arg : arguments) {
+        builder.addArgument(arg);
+      }
+    }
```

Comment:
It's not clear the message pattern this is intending to construct. A simple example in inline comments could help with debugging this.